safe/gui/tools/wizard/wizard_dialog.py

```diff
@@ -876,3 +880,29 @@ def save_current_keywords(self):
                     current_keywords['inasafe_default_values'].items()):
                 set_inasafe_default_value_qsetting(
                     self.setting, RECENT, key, value)
+
+    # Adapted from https://stackoverflow.com/a/43126946/1198772
+    def resizeEvent(self, event):
+        """Emit custom signal when the window is re-sized.
+
+        :param event: The re-sized event.
+        :type event: QResizeEvent
+        """
+        self.resized.emit()
+        return super(WizardDialog, self).resizeEvent(event)
+
+    def after_resize(self):
+        """Method after resizing the window."""
+        if self.get_current_step() == self.step_kw_fields_mapping:
+            # Use MinimumExpanding when on the Field Mapping Step
+            vertical_size_policy = QSizePolicy.MinimumExpanding
+        else:
+            vertical_size_policy = QSizePolicy.Preferred
+
+        size_policy = QSizePolicy()
+        size_policy.setHorizontalPolicy(QSizePolicy.Preferred)
+        size_policy.setVerticalPolicy(vertical_size_policy)
+        self.scrollAreaWidgetContents.setSizePolicy(size_policy)
+        # Also reset the max height. 190 is a number that make it pretty
+        max_height = self.height() - 190
+        self.scrollAreaWidgetContents.setMaximumHeight(max_height)
```

Comment:
__[flake8]__
* [W292] no newline at end of file

<!-- Comment by [SideCI](https://sideci.com) -->